lib/propolis/src/block/tracking.rs

```diff
@@ -46,18 +63,44 @@ struct TrackingEntry<T> {
 /// Track device-specific data for outstanding block [Request]s.
 impl<T> Tracking<T> {
     pub fn new(dev: Weak<dyn Device>) -> Self {
+        Self::new_raw(dev, None)
+    }
+
+    /// Create a new tracking object with a completion callback.
+    pub fn with_completion_callback(
+        dev: Weak<dyn Device>,
+        cb: impl CompletionCallback,
+    ) -> Self {
+        Self::new_raw(dev, Some(Box::new(cb)))
+    }
+
+    fn new_raw(
+        dev: Weak<dyn Device>,
+        cb: Option<Box<dyn CompletionCallback>>,
+    ) -> Self {
```

Comment:
Not required if `with_completion_callback` is eliminated.

Reply:
Removed in 8565b0c